src/Grpc.AspNetCore.Server/InterceptorCollection.cs

```diff
@@ -0,0 +1,97 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using Grpc.Core.Interceptors;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Grpc.AspNetCore.Server
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public class InterceptorCollection : IEnumerable<InterceptorRegistration>
```

Comment:
@rynowak Did you envision the interceptor registration collection this as a fully fledged `ICollection<T>` collection or something like this?

Reply:
I wanted to start off by exposing the least amount of API/functionality and add them as need arises. For example, I'm not sure there's a need for Remove. Also, currently, there's no need for the user to instantiate a new instance of InterceptorRegistration explicitly.

I'm not opposed to these changes though.

Reply:
Actually now I think about it, though strange, it might make more sense to make it an IReadonlyList<InterceptorRegistration> but with an explicit Add method.